src/main/java/seedu/duke/TaskList.java

```diff
@@ -0,0 +1,360 @@
+package seedu.duke;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Class that stores all the tasks in an arraylist.
+ */
+public class TaskList {
+    private ArrayList<Task> taskLists;
+
+    /**
+     * Initializes an instance of Tasklist.
+     *
+     * @param listOfTasks Arraylist of Tasks to be completed.
+     */
+    public TaskList(ArrayList<Task> listOfTasks) {
+        this.taskLists = listOfTasks;
+    }
+
+    /**
+     * Checks if a input for a new task is valid.
+     *
+     * @param input Name of the task.
+     * @param task Task type.
+     * @throws DukeException if empty input is given.
+     */
+    public void checkForItem(String input, String task) throws DukeException {
+        if (input.isBlank()) {
+            throw new DukeException("The description of a " + task + " cannot be empty!");
+        }
+    }
+
+    /**
+     * Marks the task as done in a list.
+     *
+     * @param userInput Task to be completed.
+     */
+    public void completeTask(String userInput) {
```

Comment:
I noticed you have the same method name in Storage, it would be more readable if the names can tell which class they belong to.